src/configure/helper/templateHelper.ts

```diff
@@ -151,6 +151,38 @@ export async function analyzeRepoAndListAppropriatePipeline(repoPath: string, re
     return templateResult;
 }
 
+export async function listResourceFilteredPipelines(azureSession: AzureSession, language: string, deployTarget: string, githubPatToken?: string): Promise<PipelineTemplate[]> {
+
+    let pipelineTemplates: PipelineTemplate[] = [];
+    let remoteTemplates: TemplateInfo[] = [];
+    try {
+        const client = await TemplateServiceClientFactory.getClient(azureSession.credentials, githubPatToken);
+        await telemetryHelper.executeFunctionWithTimeTelemetry(async () => {
+            remoteTemplates = await client.getResourceFilteredTemplates(language, deployTarget);
+        }, TelemetryKeys.ResourceFilterTemplateServiceDuration);
+        remoteTemplates.forEach((templateInfo: TemplateInfo) => {
+            const remoteTemplate: RemotePipelineTemplate = {
+                label: templateInfo.templateLabel,
+                targetType: getTargetType(templateInfo),
+                targetKind: getTargetKind(templateInfo),
+                templateType: TemplateType.REMOTE,
+                language: templateInfo.attributes.language,
+                id: templateInfo.templateId,
+                templateWeight: templateInfo.templateWeight,
+                workingDirectory: templateInfo.workingDirectory,
```

Comment:
this code is repeated in below function as well, can you create separate function for it with name convertToRemotePipelineTemplate